File: rerun-reports/02-python/20260829-032449/results/identity/data-plane/python/auth/identity-dp-python-credential-chain/python-azure-skills/baseline/generated-code/environment_detector.py
"""Detect the deployment environment from well-known host variables."""

from __future__ import annotations

import os
from enum import Enum
from typing import Mapping


class RuntimeEnvironment(str, Enum):
    DEV = "dev"
    CI = "ci"
    PRODUCTION = "production"


CI_MARKERS = (
    "TF_BUILD",
    "BUILD_BUILDID",
    "PIPELINE_WORKSPACE",
    "GITHUB_ACTIONS",
    "GITHUB_WORKSPACE",
    "GITLAB_CI",
    "CI_PROJECT_DIR",
    "JENKINS_URL",
    "CI",
)

PRODUCTION_MARKERS = (
    "IDENTITY_ENDPOINT",
    "MSI_ENDPOINT",
    "IMDS_ENDPOINT",
    "AZURE_FEDERATED_TOKEN_FILE",
    "WEBSITE_INSTANCE_ID",
    "KUBERNETES_SERVICE_HOST",
    "CONTAINER_APP_NAME",
)


def detect_environment(
    environ: Mapping[str, str] | None = None,
) -> RuntimeEnvironment:
    """Classify the current host as CI, production, or local development."""

    variables = os.environ if environ is None else environ
    if any(variables.get(name) for name in CI_MARKERS):
        return RuntimeEnvironment.CI
    if any(variables.get(name) for name in PRODUCTION_MARKERS):
        return RuntimeEnvironment.PRODUCTION
    return RuntimeEnvironment.DEV

File: retry-reports/java-second/001/20260828-114101/results/identity/data-plane/java/auth/identity-dp-java-credential-chain/java-azure-skills/azure-skill-mcp/generated-code/src/main/java/com/example/azureauth/EnvironmentDetector.java
package com.example.azureauth;

import java.util.Locale;
import java.util.Map;
import java.util.Set;

public final class EnvironmentDetector {
    private static final Set<String> CI_MARKERS = Set.of(
        "BUILD_BUILDID",
        "BUILD_SOURCESDIRECTORY",
        "SYSTEM_TEAMPROJECT",
        "SYSTEM_OIDCREQUESTURI",
        "GITHUB_ACTIONS",
        "GITLAB_CI",
        "JENKINS_URL",
        "TF_BUILD"
    );

    private static final Set<String> PRODUCTION_MARKERS = Set.of(
        "IDENTITY_ENDPOINT",
        "MSI_ENDPOINT",
        "IMDS_ENDPOINT",
        "WEBSITE_INSTANCE_ID",
        "CONTAINER_APP_NAME",
        "KUBERNETES_SERVICE_HOST",
        "AZURE_FEDERATED_TOKEN_FILE"
    );

    public DeploymentEnvironment detect() {
        return detect(System.getenv());
    }

    DeploymentEnvironment detect(Map<String, String> environment) {
        String override = environment.get("APP_DEPLOYMENT_ENVIRONMENT");
        if (hasText(override)) {
            return parseOverride(override);
        }

        if (isTrue(environment.get("CI")) || containsAny(environment, CI_MARKERS)) {
            return DeploymentEnvironment.CI;
        }

        if (containsAny(environment, PRODUCTION_MARKERS)) {
            return DeploymentEnvironment.PRODUCTION;
        }

        return DeploymentEnvironment.DEVELOPMENT;
    }

    private DeploymentEnvironment parseOverride(String value) {
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "dev", "development", "local" -> DeploymentEnvironment.DEVELOPMENT;
            case "ci", "pipeline" -> DeploymentEnvironment.CI;
            case "prod", "production" -> DeploymentEnvironment.PRODUCTION;
            default -> throw new IllegalArgumentException(
                "APP_DEPLOYMENT_ENVIRONMENT must be dev, ci, or production; got: " + value
            );
        };
    }

    private boolean containsAny(Map<String, String> environment, Set<String> names) {
        return names.stream().anyMatch(name -> hasText(environment.get(name)));
    }

    private boolean isTrue(String value) {
        return hasText(value) && !"false".equalsIgnoreCase(value) && !"0".equals(value);
    }

    private boolean hasText(String value) {
        return value != null && !value.isBlank();
    }
}
